__tests__/login/oidc/TokenRequester.spec.ts

```diff
@@ -0,0 +1,244 @@
+
+describe("TokenRequester", () => {
+  const defaultMocks = {
+    storageUtility: StorageUtilityMock,
+    issueConfigFetcher: IssuerConfigFetcherMock,
+    fetcher: FetcherMock,
+    dpopHeaderCreator: DpopHeaderCreatorMock,
+    joseUtility: JoseUtilityMock
+  };
+  function getTokenRequester(
+    mocks: Partial<typeof defaultMocks> = defaultMocks
+  ): TokenRequester {
+    return new TokenRequester(
+      mocks.storageUtility ?? defaultMocks.storageUtility,
+      mocks.issueConfigFetcher ?? defaultMocks.issueConfigFetcher,
+      mocks.fetcher ?? defaultMocks.fetcher,
+      mocks.dpopHeaderCreator ?? defaultMocks.dpopHeaderCreator,
+      mocks.joseUtility ?? JoseUtilityMock
+    );
+  }
+
+  const defaultReturnValues: {
+    storageRefreshToken: string;
+    storageIdp: string;
+    storageClientId: string;
+    storageClientSecret: string | null;
+    issuerConfig: IIssuerConfig;
+    responseBody: string;
+    jwt: Record<string, string>;
+  } = {
+    storageRefreshToken: "thisIsARefreshToken",
+    storageIdp: "https://idp.com",
+    storageClientId: "coolApp",
+    storageClientSecret: null,
+    issuerConfig: {
+      ...IssuerConfigFetcherFetchConfigResponse,
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      grantTypesSupported: ["refresh_token"]
+    },
+    responseBody: JSON.stringify({
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      id_token: "abcd",
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      access_token: "1234",
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      refresh_token: "!@#$"
+    }),
+    jwt: {
+      sub: "https://jackson.solid.community/profile/card#me"
+    }
+  };
+  function setUpMockedReturnValues(
+    values: Partial<typeof defaultReturnValues>
+  ): void {
+    defaultMocks.storageUtility.getForUser.mockResolvedValueOnce(
+      values.storageIdp ?? defaultReturnValues.storageIdp
+    );
+    defaultMocks.storageUtility.getForUser.mockResolvedValueOnce(
+      values.storageClientId ?? defaultReturnValues.storageClientId
+    );
+    defaultMocks.storageUtility.getForUser.mockResolvedValueOnce(
+      values.storageClientSecret ?? defaultReturnValues.storageClientSecret
+    );
+    const issuerConfig =
+      values.issuerConfig ?? defaultReturnValues.issuerConfig;
+    defaultMocks.issueConfigFetcher.fetchConfig.mockResolvedValueOnce(
+      issuerConfig
+    );
+    defaultMocks.fetcher.fetch.mockResolvedValueOnce(
+      (new NodeResponse(
+        values.responseBody ?? defaultReturnValues.responseBody
+      ) as unknown) as Response
+    );
+    defaultMocks.joseUtility.decodeJWT.mockReset();
+    defaultMocks.joseUtility.decodeJWT.mockResolvedValueOnce(
+      values.jwt ?? defaultReturnValues.jwt
+    );
+  }
+
+  it("Properly follows the refresh flow", async () => {
+    setUpMockedReturnValues({});
+    const TokenRefresher = getTokenRequester();
```

Comment:
I assume this local var should be named 'tokenRefresher' (lower case 'T') - but also, is this var a 'Refresher' or a 'Requester'? From the code it seems it's a token requester that just happens to expect a refresh token, which still makes it a `tokenRequester` (pedantic I know!).